src/main/java/org/mule/extension/http/api/HttpRequestAttributesBuilder.java

```diff
@@ -182,7 +182,7 @@ public HttpRequestAttributes build() {
     requireNonNull(requestUri, "Request URI cannot be null.");
     requireNonNull(localAddress, "Local address cannot be null.");
     requireNonNull(remoteAddress, "Remote address cannot be null.");
-    if (resolveMaskedRequestPath && listenerPath != null && requestPath != null) {
+    if (resolveMaskedRequestPath) {
```

Comment:
wasn't this required to resolve the masked path?

Reply:
Oh, that was just a suggestion of IntelliJ :P. You have the requireNonNull of those two properties just above, so it is redundant to ask again.